src/CheckpointMsg.chpl

```diff
@@ -128,9 +180,138 @@ module CheckpointMsg {
                      "Loaded array with metadata %s".format(mdName));
 
     }
+    updateLastCkptCompletion();
     return Msg.send(nameArg);
   }
 
+  // the smaller of those argument that are positive
+  private proc minOfPos(x, y: x.type) {
+    return if x > 0 then if y > 0 then min(x, y) else x
+                    else if y > 0 then y else 0: x.type;
```

Comment:
Can we make this simpler to read by having 4 return statements instead of just one?
```suggestion
  if x <= 0 && y <= 0 then return 0: x.type;
  if x <= 0 then return y;
  if y <= 0 then return x;
  return min(x, y);
```

Reply:
Resolved